### DISCUSSION TITLE
How to create an IPF of Dungeon Master

### DISCUSSION THREAD

**Discussion by ChristopheF** [keirf/disk-utilities#220] [General]
Hi,

I would like to create a working IPF from a Kryoflux dump of an original Dungeon Master for Amiga floppy disk (version 2.0 English).
It seems possible with disk-utilities, however I am lost about how to proceed. 
I guess I need to specify a "format" on the command line of disk-analyse but I have no clue about how to specify such a format.

Could you please walk me through the steps I need to perform?
Thanks!

**Comment by SidTheMole** [comment]
Something like "disk-analyse -f"Dungeon Master" "folder/track" Dungeon Master.ipf" would be a start. There are no digits specified in the KF stream in this case.
Something to be aware of is DM versions may differ from what d-a recognizes.

**Comment by mickstone** [comment]
I could be wrong, but I thought IPF could not create a working copy for early Dungeon Master version as it does not support the protection scheme.
Snip below from EAB
"The IPF as it is cannot handle yet the protection system used. It's not simply Weaks bits used (There are already Amiga [ games](https://eab.abime.net/showthread.php?t=112163&page=5#) in IPF using Weak Bits protections).... it requires more work on the IPF format to get the support.

https://eab.abime.net/showthread.php?t=112163&page=5

**Reply by SidTheMole** [reply]
Yes ,Keir said this about Dungeon Master on EAB.
"The diamond-shaped flux pattern for which the DM protection is infamous cannot be represented in an IPF.
Instead the generated IPF instead uses very short (1-2 bitcell) weak regions to simulate the fuzzy most-significant bit in each data byte in the protection sector.
Not a problem for an emulator to randomise one or two bits; but impossible to write to disk without being smart enough to swap back in something like the "diamond flux" trick."

**Comment by ChristopheF** [comment]
Thanks a lot @SidTheMole I have used the command line you supplied and successfully created my first IPF !
I am well aware that this is a 'hack' built in disk-analyse to produce an unofficial IPF that works fine in emulators but is not truthful to the original and cannot be written to a floppy disk.

**Comment by ChristopheF** [comment]
I have now successfully created IPF images of multiple versions of Amiga versions of Dungeon Master and Chaos Strikes Back.

@keirf , how hard would it be to add support for the Atari ST versions of DM and CSB ? They use the very same copy protection sector as the Amiga versions, however it is located in sector 7 on track 0 side 0.

Thanks for this great tool !

**Reply by keirf** [reply]
Needs a raw dump of these disks to inspect.

**Comment by ChristopheF** [comment]
You can download Kryoflux raw dumps of the following 5 floppy disks at https://transfert.free.fr/YBKm4tH (this link will expire in 30 days):
- Dungeon Master Atari ST version 1.1 English
- Dungeon Master Atari ST version 1.2 English
- Chaos Strikes Back Atari ST Game Disk version 2.0 English
- Chaos Strikes Back Atari ST Game Disk version 2.1 English
- Chaos Strikes Back Atari ST Utility Disk English

**Reply by keirf** [reply]
Thanks, I've stashed these on my Drive for future inspection

**Reply by ChristopheF** [reply]
You're welcome. Thanks to you.
It is unclear to me whether these versions would require that you write special code in disk-analyse, or if it would be possible to describe their format in a custom config file (specified with the -c command line parameter).
